### DISCUSSION TITLE
TrustRegionIteration stopping condition

### DISCUSSION THREAD

**Discussion by rjoomen** [tesseract-robotics/trajopt#348] [Q&A] [answered]
In the original trajopt paper and [in trajopt_sco](https://github.com/tesseract-robotics/trajopt/blob/master/trajopt_sco/src/optimizers.cpp#L764), the trust region loop of the optimization is exited when the trust region is below a certain threshold (`min_trust_box_size`). [In trajopt_sqp](https://github.com/tesseract-robotics/trajopt/blob/master/trajopt_optimizers/trajopt_sqp/src/trust_region_sqp_solver.cpp#L225), however, the trust region loop is run up to a maximum number of iterations ( `max_trust_region_expansions`). 

What is the reason for this change? Because I would expect something like this:
```
  int trust_region_iteration = -1;
  while (results_.box_size[0] >= params.min_trust_box_size)
  {
    trust_region_iteration++;
```

(On a side note: `max_trust_region_expansions` should be called `max_trust_region_shrinkings` or something like that, as an expansion immediately exits the loop.)

**Comment by Levi-Armstrong** [comment]
This was most likely an oversight in the original port. I have been going through it from time to time finding similar differences and fixing them. I would consider this a bug and it should be fixed.

**Comment by rjoomen** [comment, marked as answer]
Fixed by #349
